### ISSUE TITLE
Road to CalendarView

### ISSUE THREAD

**Issue by Tropix126** [Tropix126/fluent-svelte#33]
For the last week or so, ive been working on a Calendar/Datepicker component for fluent-svelte. It's being developed in an isolated repo [here](https://github.com/Tropix126/CalendarView). There are a few things that need to be ironed out, notably some a11y insights and page transitions, as well as some missing functionality. Thought I might track that here:
- [x] Complete reliance on the [`Intl.DateTimeFormat` API](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat) and no additional date libraries.
- [x] Localization quirks. (controlling first day of week)
- [x] Minimum/Maximum dates.
- [x] Multiple values/selections.
- [x] Keyboard navigation.
- [x] Option to label each item as the start of a new month.
- [x] Excluding certain dates (believe WinUI calls these "blackout" dates).
- [x] Standalone CalendarViewItem component.
- [x] Remove the temporary styling and integrate it with fluent styles.
- [x] Paging stuff and animations.

Stretches - Very likely won't happen.
- Non-gregorian calendar layouts Probably not possible due to how the `Date`s work atm.
- Eventual Temporal support? Probably shouldn't even include this due to the spec not being supported anywhere.
- Progressive enhancement. Probably won't happen since date inputs don't map to what i'm trying to accomplish here.

**Comment by Tropix126** [comment]
So everything except for the page slide animation is done. I had a prototype of it working, although it was janky when switching between pages and the other solutions aren't especially easy. I think I might look into how [svelte-calendar](https://github.com/6eDesign/svelte-calendar/issues) handles their infinite scroller, although that might come along later.

**Comment by Tropix126** [comment]
I might put a temporary animation in place for now, then track this in a separate issue.
